data-integration/utilities.py

```diff
@@ -165,18 +163,18 @@ def datasetExists(
     ds_id : int.
     dbURL : str, optional
         The default is "postgresql://test:example@localhost:5433/dataset".
-    tables : list of strings, optional
-        The default is ["datasets","spatial","data"].
     Returns
     -------
     bool
     """
+    tables = ["datasets", "spatial", "data"]
     engine = sqla.create_engine(dbURL)
     lengths = []
     for table in tables:
         with engine.connect() as con:
             rs = con.execute(
-                "SELECT COUNT(*) FROM {} WHERE ds_id = {}".format(table, ds_id)
+                "SELECT COUNT(*) FROM {} WHERE ds_id = %(ds_id)s;".format(table),
```

Comment:
@gcmalloc any suggestion on how to safely (and simply) pass the table name to the query with sqlalchemy? The Pythonic interface of sqlalchemy seems an overkill here. Anyway, I provisionally set the table list inside the function, so this should also be safe enough, right?

Reply:
You could use 

engine.has_table(table) as a check, you could also use engine.table_names() to get the list of tables. You could also call the underlying escape of psql (sql.Identifier see https://stackoverflow.com/questions/13793399/passing-table-name-as-a-parameter-in-psycopg2). Another solution would be to rewrite the function to directly include the table and make either an union of all matches or add the three count from the separate tables.

Alternatively, you can use sqlalchemy function directly, but not using the ORM feature is mostly getting in your way there. Maybe you want to use https://www.psycopg.org/ directly instead ?

Reply:
Thank you @gcmalloc ! I ended up using psycopg2 directly, as it was already installed. I think it is much cleaner (and safer) now.
@melhk Thanks, I updated the description as well